File: clasespersona/clase.php
<?php
class Persona {
    protected $nombre;
    protected $edad;

    public function __construct($nombre, $edad) {
        $this->nombre = $nombre;
        $this->edad = $edad;
    }

    public function __toString() {
        return $this->nombre . ", " . $this->edad . " años";
    }
}

class Gerente extends Persona {
    private $responsabilidad;

    public function __construct($nombre, $edad, $responsabilidad) {
        parent::__construct($nombre, $edad);
        $this->responsabilidad = $responsabilidad;
    }

    public function __toString() {
        return parent::__toString() . ", Gerente responsable de " . $this->responsabilidad;
    }
}

class Dueno extends Persona {
    private $propiedad;

    public function __construct($nombre, $edad, $propiedad) {
        parent::__construct($nombre, $edad);
        $this->propiedad = $propiedad;
    }

    public function __toString() {
        return parent::__toString() . ", Dueño de " . $this->propiedad;
    }
}

class Administrativo extends Persona {
    private $departamento;

    public function __construct($nombre, $edad, $departamento) {
        parent::__construct($nombre, $edad);
        $this->departamento = $departamento;
    }

    public function __toString() {
        return parent::__toString() . ", Administrativo del departamento de " . $this->departamento;
    }
}

class Docente extends Persona {
    private $materia;

    public function __construct($nombre, $edad, $materia) {
        parent::__construct($nombre, $edad);
        $this->materia = $materia;
    }

    public function __toString() {
        return parent::__toString() . ", Docente de " . $this->materia;
    }
}

class Maestranza extends Persona {
    private $tareas;

    public function __construct($nombre, $edad, $tareas) {
        parent::__construct($nombre, $edad);
        $this->tareas = $tareas;
    }

    public function __toString() {
        return parent::__toString() . ", Maestranza encargado de " . $this->tareas;
    }
}

File: clasespersona/clase.py
class Persona:
    def __init__(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad

    def __str__(self):
        return f'{self.nombre}, {self.edad} años'

class Gerente(Persona):
    def __init__(self, nombre, edad, responsabilidad):
        super().__init__(nombre, edad)
        self.responsabilidad = responsabilidad

    def __str__(self):
        return f'{self.nombre}, {self.edad} años, Gerente responsable de {self.responsabilidad}'

class Dueño(Persona):
    def __init__(self, nombre, edad, propiedad):
        super().__init__(nombre, edad)
        self.propiedad = propiedad

    def __str__(self):
        return f'{self.nombre}, {self.edad} años, Dueño de {self.propiedad}'

class Administrativo(Persona):
    def __init__(self, nombre, edad, departamento):
        super().__init__(nombre, edad)
        self.departamento = departamento

    def __str__(self):
        return f'{self.nombre}, {self.edad} años, Administrativo del departamento de {self.departamento}'

class Docente(Persona):
    def __init__(self, nombre, edad, materia):
        super().__init__(nombre, edad)
        self.materia = materia

    def __str__(self):
        return f'{self.nombre}, {self.edad} años, Docente de {self.materia}'

class Maestranza(Persona):
    def __init__(self, nombre, edad, tareas):
        super().__init__(nombre, edad)
        self.tareas = tareas

    def __str__(self):
        return f'{self.nombre}, {self.edad} años, Maestranza encargado de {self.tareas}'

